### PR TITLE
updated deprecated packages

### PR DESCRIPTION
**Pull request by majesticio**
Adapter now updated
Changes:
- request -> got
- moment -> luxon
- Number -> parseFloat

### CODE REVIEW

File: src/adapters/arpalazio.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import log from '../lib/logger.js';
 import { acceptableParameters, convertUnits } from '../lib/utils.js';
 import { default as baseRequest } from 'request';
 import { default as rp } from 'request-promise-native';
-import { default as moment } from 'moment-timezone';
+import { DateTime } from 'luxon';
 import sj from 'scramjet';
 const { StringStream, MultiStream } = sj;
```
Comment (russbiggs) [inline src/adapters/arpalazio.js:11]: move this below imports

File: src/adapters/arpalazio.js [hunk 2]
```diff
@@ -113,7 +113,7 @@ export const getStream = function (cityName, url, averagingPeriod, source, orgUr
   const unit = getUnit(parameter);
   const dayPosition = averagingPeriod.value === 1 ? 0 : 1;
 
-  const fewDaysAgo = +Number(moment.tz(timezone).subtract(4, 'days').format('DDD'));
+  const fewDaysAgo = +parseFloat(DateTime.local().setZone(timezone).minus({days: 4}).ordinal);
```
Comment (russbiggs) [inline src/adapters/arpalazio.js:116]: Why do we check 4 days ago?

File: src/adapters/arpalazio.js [hunk 3]
```diff
@@ -141,11 +141,12 @@ export const getStream = function (cityName, url, averagingPeriod, source, orgUr
...
-          ? moment.tz(`${year} ${date1} ${date2}`, 'YYYY DDD HH', timezone)
-          : moment.tz(`${year} ${date2}`, 'YYYY DDD', timezone);
+          ? DateTime.fromObject({year, ordinal: date1, hour: date2}).setZone(timezone)
+          : DateTime.fromObject({year, ordinal: date2}).setZone(timezone);
+
         const date = {
-          utc: timestamp.toDate(),
-          local: timestamp.format()
+          utc: timestamp.toUTC().toFormat("yyyy-MM-dd'T'HH:mm:ss'Z'"),
+          local: timestamp.toFormat("yyyy-MM-dd'T'HH:mm:ssZZ")
```
Comment (russbiggs) [inline src/adapters/arpalazio.js:149]: Why is Z quoted in the line above but not here?

**Review by russbiggs — CHANGES_REQUESTED** [review]

File: src/adapters/arpalazio.js (hunk 2 above)
Comment (majesticio) [reply src/adapters/arpalazio.js]: I will create an issue to audit this adapter.

File: src/adapters/arpalazio.js (hunk 3 above)
Comment (majesticio) [reply src/adapters/arpalazio.js]: I have found it useful to explicitly set the format, I am open to suggestions.

**Comment by majesticio** [comment]
merged into update-libraries

**Pull request closed without merging**